### ISSUE TITLE
Whisper Turbo model run on a CPU

### ISSUE THREAD

**Issue by joseph16388** [SYSTRAN/faster-whisper#1144]
Hello！
How fast does the Whisper Turbo model run on a CPU?
thanks!

**Comment by joseph16388** [comment]
> Should be a bit slower than medium I think.

just a bit slower？
Previous large model ran much slower on CPUs compared to GPUs, by a factor of 5-10 or so.

**Comment by Purfview** [comment]
> just a bit slower？

That's what I wrote.

>Previous large model ran much slower on CPUs compared to GPUs, by a factor of 5-10 or so.

That's expected.

**Comment by DakeQQ** [comment]
Feel free to reference this [repository](https://github.com/DakeQQ/Transcribe-and-Translate-Subtitles) — it's designed for users with CPU-only. With an Intel i3-12300 CPU and Whisper-Large-V3-Turbo, it takes just 15 minutes to generate subtitles for a 2-hour movie. Additionally, the faster SenseVoiceSmall model can transcribe the same movie in just 7 minutes. This tool also includes a VAD and denoiser to improve subtitle accuracy.

**Comment by FerLuisxd** [comment]
@DakeQQ with the same quality? what about the vram usage?

**Comment by DakeQQ** [comment]
@FerLuisxd 
VRAM typically refers to GPU memory. However, this repository currently does not support GPU acceleration and is designed for users with CPU-only setups seeking faster transcription performance. A computer with 16GB of RAM can run Denoiser, VAD, Whisper-Large-V3, and a 7B LLM for translation simultaneously, making it highly efficient for light hardware setups.

We haven't tested the optimized model's output for CER or WER yet, but the generated subtitles are accurate enough for watching movies, even with a few minor errors.

**Comment by Purfview** [comment]
> Feel free to reference this [repository](https://github.com/DakeQQ/Transcribe-and-Translate-Subtitles) — it's designed for users with CPU-only. With an Intel i3-12300 CPU and Whisper-Large-V3-Turbo, it takes just 15 minutes to generate subtitles for a 2-hour movie. Additionally, the faster SenseVoiceSmall model can transcribe the same movie in just 7 minutes. This tool also includes a VAD and denoiser to improve subtitle accuracy.

```
  File "d:\Programs\Python_3.10.11\lib\shutil.py", line 254, in copyfile
    with open(src, 'rb') as fsrc:
FileNotFoundError: [Errno 2] No such file or directory: './modeling_modified/utils_vad.py'
```

**Comment by DakeQQ** [comment]
@Purfview 
Please download the necessary models from [Google Drive](https://drive.google.com/file/d/1RI8dIh7oQgonTW-A1dR7bJgsYcIjYybH/view?usp=drive_link). After downloading, unzip the file, and you will see the following contents. This issue should be solved.

![image](https://github.com/user-attachments/assets/3feeae95-b59a-4764-bbf0-66accf070b77)

**Comment by Purfview** [comment]
Oh, I didn't copied those from the main folder.

Quick performance "benchmark":
Tested only turbo,
Transcription took the same time as sequential faster-whisper, so in batched mode FW would be much faster for me,
~60% of text is missing.
Timestamps are super inaccurate.

**Comment by DakeQQ** [comment]
Dear Sir/Madam,  

Thank you for sharing your observations. I'd like to clarify a few key points to avoid any misunderstandings:  

1. **Silero-VAD and Faster-Whisper**  
We use **the official Silero-VAD model**, which performs identically to Faster-Whisper when using the same parameters. Therefore, **identical timestamp outputs** should be expected with matching configurations. Major differences in transcription quality are likely due to **incorrect parameter settings**.  

2. **Performance Comparison**  
Your benchmark may be **misleading**. Our time measurements **include the entire process** — video loading, model loading, caching, and subtitle generation — not just Whisper inference time. Comparing only inference times is **not a fair comparison**.  

3. **Accuracy and Missing Text**  
Missing text or inaccurate timestamps are typically caused by **misconfiguration**. Disabling VAD does not result in more text being transcribed; instead, it may capture **irrelevant noise**, **increasing processing time and reducing accuracy**. We use the official Whisper-Large-V3 model optimized with ONNX Runtime. While minor accuracy differences may occur, they should align closely with Faster-Whisper when properly configured.  
We also offer multiple **denoiser** that works alongside VAD to filter out background noise, enhancing transcription accuracy.  

4. **Batch Mode Speed**  
Our solution is **not slower than Faster-Whisper** in CPU batch processing. Performance depends on your **hardware setup** and **parameter configurations**.  

In summary, the tool’s performance and accuracy are influenced by **how it’s configured**. The issues you've reported are likely due to **misconfiguration** rather than model limitations.  

Please let me know if you need help optimizing your settings for the best results.  

Best regards

**Comment by Purfview** [comment]
>We use the official Silero-VAD model, which performs identically to Faster-Whisper when using the same parameters. 

I disabled VAD to see how it performs in sequential mode.

>Major differences in transcription quality are likely due to incorrect parameter settings.

There is literally no parameters to adjust, just to select a different language.

>Your benchmark may be misleading. 

Yeah it was. FW was measured with whole python+model loading vs ASR time for your repo.

>Missing text or inaccurate timestamps are typically caused by misconfiguration. 

There is nothing to configure.

>Disabling VAD does not result in more text being transcribed; instead, it may capture irrelevant noise, increasing processing time and reducing accuracy.

I didn't disabled it, ran it with defaults, but now ~90% of text is missing instead of ~70%... and timestamps are OK now.

>Our solution is not slower than Faster-Whisper in CPU batch processing. 

Indeed it's 1.4 times faster in batched.

>The issues you've reported are likely due to misconfiguration rather than model limitations.

I'm sure that missing text is implementation issue.

**Comment by DakeQQ** [comment]
Thank you for your reply. We have verified the V3-Turbo model with voice segments in Chinese, English, Japanese, and Korean, all yielding correct outputs. Therefore, we are uncertain about the cause of the poor experience you've encountered. If you're working on an English transcription task, we recommend using **SenseVoiceSmall** or **Paraformer-Large**, and enabling a denoiser. We believe this setup will enhance subtitle quality and processing speed.

Best regards.

**Comment by DakeQQ** [comment]
Based on our test results, we believe that it may not be entirely accurate to compare SenseVoiceSmall and Paraformer-Large with Whisper Small and Tiny. While we observed some errors in the generated subtitles, it's important to highlight the key advantage of these models: their processing time is shorter than that of Whisper. As the saying goes, it is difficult to achieve both perfection and speed at the same time.

---
`Whisper-Large-V3-Turbo`
---

WEBVTT

0
00:00:00.000 --> 00:00:03.000
 You're only inspired yet?

1
00:00:08.099 --> 00:00:10.800
 Thank you.

2
00:00:13.900 --> 00:00:16.771
 There's a boat tour of the Trakla Island formations this afternoon.

3
00:00:16.771 --> 00:00:19.642
 I was thinking we could go on that and make reservations in town for dinner.

4
00:00:19.642 --> 00:00:22.514
 We could try the Chinese place.

5
00:00:22.514 --> 00:00:25.385
 I don't think I'd survive another dinner in town.

6
00:00:25.385 --> 00:00:28.257
 Even the idea of it.

7
00:00:28.257 --> 00:00:31.128
 Does anyone think it's

8
00:00:37.100 --> 00:00:38.399
,a

9
00:00:44.100 --> 00:00:49.100
 Is it okay if I go? I'll meet you on the bridge. No? No?

---

`SenseVoice`
---

WEBVTT

0
00:00:00.000 --> 00:00:00.400
The.

1
00:00:01.700 --> 00:00:03.000
I inspired yet.

2
00:00:08.099 --> 00:00:08.599
Thank you.

3
00:00:09.900 --> 00:00:10.800
Is that you were hungry?

4
00:00:13.900 --> 00:00:17.600
There's a boat tour of the Trakler Island formations this afternoon.

5
00:00:18.899 --> 00:00:20.899
I was thinking we could go on that.

6
00:00:21.399 --> 00:00:23.199
Make reservations in town for dinner.

7
00:00:24.600 --> 00:00:25.899
You could try the Chinese place.

8
00:00:26.600 --> 00:00:28.399
I don't think I survived another dinner in town.

9
00:00:30.600 --> 00:00:31.600
Even the idea of that.

10
00:00:32.700 --> 00:00:34.000
Does anyone to get to real town?

11
00:00:37.100 --> 00:00:38.399
What would have a Chinese place.

12
00:00:44.100 --> 00:00:45.100
 okay if I go.

13
00:00:45.399 --> 00:00:46.399
I'll meet you on which.

14
00:00:48.000 --> 00:00:48.299
I.

---

`Paraformer-Large`
---

WEBVTT

0
00:00:00.000 --> 00:00:00.400
mmmm mmhmm

1
00:00:01.700 --> 00:00:03.000
really inspired yet

2
00:00:08.099 --> 00:00:08.500
thank you

3
00:00:10.000 --> 00:00:10.800
so you were hungry

4
00:00:13.900 --> 00:00:17.699
as a art tour of the trackler island formations this afternoon

5
00:00:18.899 --> 00:00:23.100
i was thinking we could go on that and make reservations in town for dinner

6
00:00:24.600 --> 00:00:28.399
could could try the chinese place i think think survisurvion another dinner and sound

7
00:00:30.600 --> 00:00:34.000
even the idea that does anyone want to get a real town

8
00:00:37.100 --> 00:00:38.500
i would have a chinese place

9
00:00:44.100 --> 00:00:46.299
say k if i go i'll make you on the beach

10
00:00:48.000 --> 00:00:49.100
no true

---

**Comment by Purfview** [comment]
>Whisper-Large-V3-Turbo

Your result looks OK [except the 8th line], for me with this audio only one line is properly transcribed, at other timestamps there is gibberish like in your result at "8" line.

Results from other models look mine results, I think.

**Comment by DakeQQ** [comment]
We greatly appreciate your testing. We discovered that the issue with our V3-Turble was caused by incorrect VAD default parameters. The `run.py` script has now been updated, and you can use the default values to achieve the following results.

There is no doubt that Faster-Whisper performs better, and we will continue working to improve our tool to reach its level. 

Best regards.

Whisper-Large-V3-Turbo
---
WEBVTT

0
00:00:00.000 --> 00:00:03.000
 Be feeling inspired yet?

1
00:00:08.099 --> 00:00:10.800
 Thank you.

2
00:00:13.900 --> 00:00:16.800
 There's a boat tour of the Trakla Island formations this afternoon.

3
00:00:16.800 --> 00:00:19.699
 I was thinking we could go on that and make reservations in town for dinner.

4
00:00:19.699 --> 00:00:22.600
 We could try the Chinese place.

5
00:00:22.600 --> 00:00:25.500
 I don't think I survive another dinner in town

6
00:00:30.600 --> 00:00:34.000
 Even the idea, man. Does anyone think it's a real town?

7
00:00:37.100 --> 00:00:38.500
 A.

8
00:00:44.100 --> 00:00:49.100
 Is it okay if I go? I'll meet you on the bridge. No? No. No.

---
